astropy/modeling/core.py

```diff
@@ -1535,49 +1562,59 @@ def _validate_input_units(self, inputs, equivalencies=None):
                     # We check for consistency of the units with input_units,
                     # taking into account any equivalencies
 
-                    if inputs[i].unit.is_equivalent(input_unit, equivalencies=input_units_equivalencies[input_name]):
+                    if inputs[i].unit.is_equivalent(
+                            input_unit,
+                            equivalencies=input_units_equivalencies[input_name]):
 
                         # If equivalencies have been specified, we need to
-                        # convert the input to the input units - this is because
-                        # some equivalencies are non-linear, and we need to be
-                        # sure that we evaluate the model in its own frame
-                        # of reference. If input_units_strict is set, we also
-                        # need to convert to the input units.
-                        if len(input_units_equivalencies) > 0 or self.input_units_strict[input_name]:
-                            inputs[i] = inputs[i].to(input_unit, equivalencies=input_units_equivalencies[input_name])
+                        # convert the input to the input units - this is
+                        # because some equivalencies are non-linear, and
+                        # we need to be sure that we evaluate the model in
+                        # its own frame of reference. If input_units_strict
+                        # is set, we also need to convert to the input units.
+                        if len(input_units_equivalencies) > 0 or \
+                               self.input_units_strict[input_name]:
```

Comment:
We should avoid ``\`` and use brackets instead for multi-line statements (also just below)